src/odemis/gui/comp/legend.py

```diff
@@ -365,3 +367,93 @@ def on_paint(self, _):
             self._max_tick_width = max_width
             self.SetMinSize((self._max_tick_width + 14, -1))
             self.Parent.GetSizer().Layout()
+
+
+class RadioLegend(wx.Panel):
+    """
+    This class describes a legend containing radio buttons,
+    where each button represents a possible selection of a VA.
+    """
+
+    def __init__(self, parent, wid=-1, pos=(0, 0), size=wx.DefaultSize, style=wx.NO_BORDER):
+        wx.Panel.__init__(self, parent, wid, pos, size, style)
+
+        self.SetBackgroundColour(parent.GetBackgroundColour())
+        self.SetForegroundColour(parent.GetForegroundColour())
+
+        # get colors: if ar-viewport is selected, to avoid colors for bg and fg for txt will be taken from selection
+        self.bg_color = parent.GetBackgroundColour()
+        self.fg_color = parent.GetForegroundColour()
+
+        # descriptive VA text
+        self.text = wx.TextCtrl(self, style=wx.NO_BORDER | wx.CB_READONLY)
+        self.text.SetBackgroundColour(parent.GetBackgroundColour())
+        self.text.SetForegroundColour(parent.GetForegroundColour())
+        self.text.SetToolTipString("Polarization direction currently displayed")
+        self.text.SetValue("Polarization:")
+
+        self.pol_display = None
+        self.control_sizer = wx.BoxSizer(wx.HORIZONTAL)
+
+        # legend_panel_sizer is needed to add a border around the legend
+        self.border_sizer = wx.BoxSizer(wx.VERTICAL)
+        self.border_sizer.Add(self.control_sizer, border=8, flag=wx.ALL | wx.EXPAND)
+
+        self.SetSizer(self.border_sizer)
+
+    def clear(self):
+        self.Refresh()
+
+    @wxlimit_invocation(0.2)
+    def Refresh(self):
+        """
+        Refresh, which can be called safely from other threads
+        """
+        wx.Panel.Refresh(self)
+
+    def changePos(self, event):
```

Comment:
In the GUI we try to "only" have two conventions for the function names. Either "FunctionName" or "function_name" (and nothing like "functionName"). Usually, we use "FunctionName" for everything wx-related, as it's the wx convention, and "function_name" for other functions. I'd recommend to name it `OnPolarizationButton()`.